### PR TITLE
test(azure): enabled fast azure provision tests

### PR DESCRIPTION
**Pull request by soyacz**
Currently Azure provisioning tests are disabled because they involved real Azure service.
This commit makes tests to use fake Azure service that operates on files instead of real resources.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: unit_tests/provisioner/test_azure_provisioner.py [hunk 1]
```diff
@@ -13,16 +13,25 @@
...
+from unit_tests.provisioner.fake_azure_service import FakeAzureService  # pylint: disable=import-error
 
 
-class DisabledTestProvisionScyllaInstanceAzureE2E:
+class TestProvisionScyllaInstanceAzureE2E:
     """this is rather e2e test - takes around 8 minutes (2m provisioning, 6 min cleanup with wait=True)"""
 
+    @pytest.fixture(scope="session")
+    def azure_service(self, tmp_path_factory) -> AzureService:  # pylint: disable=no-self-use
+        if False:  # make it True to test with real Azure pylint: disable=using-constant-test
```
Comment (fgelcer) [inline unit_tests/provisioner/test_azure_provisioner.py:30]: it would be more clear if we do:
```
run_on_real_azure = False
if run_on_real_azure:
```

**Review by fgelcer — COMMENTED** [review]
only 1 single comment, other than that, LGTM

**Review by fgelcer — APPROVED** [review]
LGTM

File: unit_tests/provisioner/test_azure_provisioner.py (hunk 1 above)
Comment (fgelcer) [reply unit_tests/provisioner/test_azure_provisioner.py]: thank you

**Comment by fgelcer** [comment]
@soyacz , there is a pylint failure, could you please look into it and send a fix for it?

**Comment by soyacz** [comment]
> @soyacz , there is a pylint failure, could you please look into it and send a fix for it?

Some instability in pylint? I rebased (no code affected) and pre-commit passed.

**Review by fgelcer — APPROVED** [review]
LGTM

**Comment by soyacz** [comment]
@enaydanov @bentsi ping

**Comment by soyacz** [comment]
@bentsi @enaydanov ping

**Review by enaydanov — APPROVED** [review]

**Pull request merged**